### DISCUSSION TITLE
Duplicate Key模型可以根据排序字段去重吗？UNIQUE KEY模型可以去掉读时合并的的策略不？

### DISCUSSION THREAD

**Discussion by ltylty** [apache/doris#9989] [A - General / Q&A]
Duplicate Key模型可以根据排序字段去重吗？ 或者 UNIQUE KEY模型可以去掉读时合并的的策略不？ 由我自己控制什么时候合并。

我介绍一下我的使用背景。
之前一直使用clickhouse，但join性能比较差，着手调研doris。

我的使用场景是flink实时任务将明细数据写入doris，为了极致的查询性能（不能慢于clickhouse）需要选用UNIQUE KEY模型。
但是我没法保证flink写入starrocks的EXACTLY_ONCE，只能保证at least once。有时候flink实时任务需要重启也会导致数据重复写入。
所以需要在数据仓库层支持幂等性写入。但是明细层并不需要实时地进行数据去重，发现数据有问题之后手动进行去重也可以接受。

之前用clickhouse的replacingMergeTree可以比较方便的解决这个问题。

**Comment by zy-kkk** [comment]
你好，Doris在即将发布的1.2版本中支持了merge on write的unique key模型，相比旧的unique key有很大的性能提升：[merge on write unique key](https://doris.apache.org/zh-CN/docs/dev/data-table/datamodel#%E5%86%99%E6%97%B6%E5%90%88%E5%B9%B612%E7%89%88%E6%9C%AC%E6%96%B0%E5%A2%9E)
关于2pc写入，目前的[doris-flink-connector](https://doris.apache.org/zh-CN/docs/dev/ecosystem/flink-doris-connector)已经支持

**Comment by ketingli1** [comment]
@ltylty 您好，您是怎么解决的

**Reply by ltylty** [reply]
> @ltylty 您好，您是怎么解决的

用 clickhouse  😔

**Reply by ketingli1** [reply]
为什么不用更新表

**Reply by ketingli1** [reply]
或者主键表 @ltylty

**Reply by ltylty** [reply]
不管是 merge on write 还是 merge on read 都是需要成本的。正常情况我不需要merge，一些特殊场景需要手动merge。
clickhouse的replacingMergeTree可以完美解决这个问题。
